### PR TITLE
Add documentation for the transport package.

### PR DESCRIPTION
**Pull request by kisom**
This also updates the authentication docs to describe the various ways to provide the authentication key.

### CODE REVIEW

File: doc/transport.txt [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+
+Transport package communication models
+--------------------------------------
+
+There are three models for communications:
+
+1. A general TLS listener, such as a public HTTPS server. In this
+   model, the server does not expect clients to authenticate
+   themselves to the server using client authentication
+   certificates. This listener doesn't need to configure any
```
Comment (Bren2010) [inline doc/transport.txt]: Unfinished

**Comment by Bren2010** [comment]
If these are all in markdown, why are they `.txt`? GitHub will render them if they're `.md`.

**Comment by kisom** [comment]
They're not in markdown.

File: doc/transport.txt (hunk 1 above)
Comment (kisom) [inline doc/transport.txt]: Updated, looks like a line was cut off during editing.

**Comment by lziest** [comment]
lgtm

**Pull request merged**